### ISSUE TITLE
[kernel] test whether the kernel is batch variant or not

### ISSUE THREAD

**Issue by aolemila** [sgl-project/sglang-jax#660]
# Introduction
This issue will record the batch invariance of JAX native kernels and Pallas kernels.

## Environment

- Hardware: tpu-v5p
- Software:
```bash
flax                         0.12.2
jax                          0.8.1
jaxlib                       0.8.1
jaxtyping                    0.3.6
libtpu                       0.0.30
```

Note: Why do we select v5p? Because v5p chip has two tensor cores and it is relatively easy to get compared with v7.
## Kernel Collection

Based on the implementation of qwen3 dense model(`qwen3.py`), here is a list of kernels we firstly focus on.

- `lax.dot_general`: batch variant, note bias add
- `jnp.dot`: batch variant
- [Optional] `jnp.einsum`: current usage can be replaced by matmul: "n,d->nd", "nd,md->nm"
- `jax._src.nn.log_softmax`: jax._src.nn.log_softmax(logits, axis=-1), x-max(x)
- `jax.nn.softmax`: jax.nn.softmax(processed_logits, axis=-1), x-max(x)
- `jnp.sum`: jnp.sum(p, axis=1, keepdims=True)
- `jnp.cumsum`: jnp.cumsum(probs_sort, axis=-1)
- `x.mean(axes, where=mask)`
- `lax.pmean`: lax.pmean(mus[0], axis_name, axis_index_groups=axis_index_groups)
- `jax.vmap`: jax.vmap(unpack_batch_item)(vocab_mask), unpack_batch_item is element-wise

## Scenarios
- One chip(multi tensor cores) per Host/VM
- Multi chips per Host/VM
- Multi hosts in the Pod
